internal/provider/networks_devices_claim_resource.go

```diff
@@ -0,0 +1,234 @@
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to create resource",
+			fmt.Sprintf("%v\n", err.Error()),
+		)
+	}
+
+	// Collect any HTTP diagnostics that might be useful for debugging.
+	if httpResp != nil {
+		tools.CollectHttpDiagnostics(ctx, &resp.Diagnostics, httpResp)
+	}
+
+	// If it's not what you expect, add an error to diagnostics.
+	if httpResp.StatusCode != 200 {
+		resp.Diagnostics.AddError(
+			"Unexpected HTTP Response Status Code",
+			fmt.Sprintf("%v", httpResp.StatusCode),
+		)
+	}
+
+	// If there were any errors up to this point, log the plan data and return.
+	if resp.Diagnostics.HasError() {
+		resp.Diagnostics.AddError("Plan Data", fmt.Sprintf("\n%v", data))
+		return
+	}
+
+	// Set ID for the new resource.
+	data.Id = jsontypes.StringValue("example-id")
+
+	// Now set the final state of the resource.
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+
+	// Log that the resource was created.
+	tflog.Trace(ctx, "created resource")
+}
+
+// Read method is responsible for reading an existing resource's state.
+// It takes a ReadRequest and returns a ReadResponse with the current state of the resource or an error.
+func (r *NetworksDevicesClaimResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data *NetworksDevicesClaimResourceModel
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	// Check if there are any errors before proceeding.
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Set ID for the resource.
+	data.Id = jsontypes.StringValue("example-id")
+
+	// Now set the final state of the resource.
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+
+	// Log that the resource was read.
+	tflog.Trace(ctx, "read resource")
+}
+
+// Update function is responsible for updating the state of an existing resource.
+// It uses an UpdateRequest and responds with an UpdateResponse which contains the updated state of the resource or an error.
+func (r *NetworksDevicesClaimResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	var data *NetworksDevicesClaimResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	// If there was an error reading the plan, return early.
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Set ID for the resource.
+	data.Id = jsontypes.StringValue("example-id")
+
+	// Now set the updated state of the resource.
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+
+	// Log that the resource was updated.
+	tflog.Trace(ctx, "updated resource")
+}
+
+// Delete function is responsible for deleting a resource.
+// It uses a DeleteRequest and responds with a DeleteResponse which contains the updated state of the resource or an error.
+func (r *NetworksDevicesClaimResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
```

Comment:
So there is a API call to delete a single device: {{baseUrl}}/networks/:networkId/devices/remove

What we can do here is loop through data.serials and make that API call, then remove from state.

Reply:
You should also be able to delete a network which contains serials without issue but when it comes to the organization there isn't a straightforward way to unclaim it. We may have to use the organization claim resource to move it intio our static "DO_NOT_DELETE" org so we can delete the ephemeral test_acc organization made during the tests:

https://community.meraki.com/t5/Developers-APIs/Unclaim-devices-from-an-ORG-API/m-p/76010/highlight/true

Reply:
Understand I added accordingly to move the stuff to DO_NOT_DELETE org id